src/util/TraceUtil.v3

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 Wizard Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+component TraceUtil {
+    // Renders a right-aligned string in a column of length `len`
+    def renderRight(out: TraceBuilder, len: int, f: TraceBuilder -> void) {
+        var str = TraceBuilder.new();
+        f(str);
+        for (i < len - str.length) {
+            out.putc(' ');
+        }
+        out.puts(str.extract());
+    }
+    // Renders the decimal `dividend/divisor` to `prec` decimal places
+    def renderFrac(out: TraceBuilder, dividend: long, divisor: long, prec: int) {
+        var intVal = dividend / divisor;
+        out.putd(intVal);
+        out.putc('.');
+
+        var base = 10;
+        for (i < prec) {
+            out.putd((dividend * base / divisor) % 10);
+            base *= 10;
+        }
+    }
+    def renderPct(out: TraceBuilder, dividend: long, divisor: long, prec: int) {
```

Comment:
I think I prefer spelling this name out: `renderPercent`.

I'm a little inconsistent on this myself, so it'd probably be good for me to develop some kind of guideline on naming.